src/types.js

```diff
@@ -95,6 +95,20 @@ export type Account = {|
    */
   zulipVersion: ZulipVersion | null,
 
+  /**
+   * An integer indicating what features are available on the server.
+   *
+   * The feature level increases monotonically; a value of N means the
+   * server supports all API features introduced before feature level N.
+   * This is designed to provide a simple way for mobile apps to decide
+   * whether the server supports a given feature or API change.
+   *
+   * Like zulipVersion, we learn the feature level from /server_settings
+   * at the start of the login process, and again from /register when
+   * setting up an event queue.
+   */
+  zulipFeatureLevel: number | null,
```

Comment:
**edit**: Hmm, I'm actually not sure about what I say below the line (below); @gnprice said, at https://github.com/zulip/zulip-mobile/pull/4079#discussion_r418363632, that the `null` does serve a purpose:

> We'll still need the type to be `number | null`, I think, to handle accounts that the user hasn't looked at since upgrading to an app version that stores feature levels.

I think maybe it's clearer if we do forget about `null`, and just make a migration that sets it to `0` if it's not there? Maybe not, or maybe I'm missing something. 🙂 What do you think, Greg?

-----

This should also be just `number`, to add to the changes you've made in response to https://github.com/zulip/zulip-mobile/pull/4079#discussion_r421867948.

Correspondingly, the `zulipFeatureLevel` param in `makeAccount` in exampleData.js should be `zulipFeatureLevel?: number`. The `?` is so that we can call `makeAccount` without deciding what `zulipFeatureLevel` we want; the implementation can pick one.

Reply:
**edit**: See above.

A few places still have `zulipFeatureLevel: null` being set explicitly (you can search for these); they should be replaced with `zulipFeatureLevel: 0`.

Reply:
> I think maybe it's clearer if we do forget about `null`, and just make a migration that sets it to `0` if it's not there?

The thing about that is that it might not be true -- the server might actually be on a later version, and we just haven't talked to it since upgrading the app to a version that stores the feature level.

In particular, anyone using the app now with chat.zulip.org or zulip.com has accounts where we've recorded that the version is 3.0-rc1-something. There is no server (unless a willfully confusing one) that has that version number and a feature level of 0.

Reply:
Ah, right—thanks for reminding me about that possibility. So my comments here should be disregarded, except:

I think maybe we'll need a migration that sets it to `null` (otherwise it'll be missing altogether)? Something like:

```javascript
  // Accounts.zulipFeatureLevel is `number | null`
   '14': state => ({
    ...state,
    accounts: state.accounts.map(a => ({
      ...a,
      zulipFeatureLevel: a.zulipFeatureLevel !== undefined ? a.zulipFeatureLevel : null,
    })),
  }),
```

Reply:
Yeah, that looks right. We do need a migration, or else the `zulipFeatureLevel` property would be simply missing (contrary to what the type says.)

The code can be tightened a bit by using `??`:
```
  zulipFeatureLevel: a.zulipFeatureLevel ?? null
```

... Actually, better: it's impossible for `zulipFeatureLevel` to be present at this point. This migration runs only when we're processing data that was stored by an older version of the app that didn't know anything about `zulipFeatureLevel` here. So we can just say:
```
  zulipFeatureLevel: null
```
and that's clearer about about what's happening because it doesn't try to handle a situation that should be impossible.

(Looking back at previous migrations, I see we did the redundant-conditional thing with the last two, oops.)